src/org/dux/stracetool/StraceParser.java

```diff
@@ -0,0 +1,54 @@
+package org.dux.stracetool;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+// TODO What to do about DuxCLI references (primarily debug statements)?
+import org.dux.cli.DuxCLI;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class StraceParser {
+    public static List<StraceCall> parse(String path)
+            throws IOException, FileNotFoundException {
+        String os = System.getProperty("os.name");
+        StraceParser sp = null;
+        if (os.startsWith("Linux")) {
+            sp = new LinuxStraceParser();
+        } else if (os.startsWith("Windows")) {
+            // TODO implement
+        } else {
+            return null;
```

Comment:
I'd rather throw an exception than return `null` here (and above, where Windows isn't implemented). That's a cleaner way to fail.